airbnb_project/listings/harvester_app/harvester/spiders/listings_spider.py

```diff
@@ -0,0 +1,312 @@
+    name = "listings_spider"
+
+    # List of allowed domains for the spider to crawl
+    allowed_domains = ['airbnb.ca']
+
+    # List to store pagination cursors for navigating through result pages
+    next_page_cursors: [str] = None
+
+    # Counter to keep track of the total number of listings scraped
+    total_listings = 0
+
+    # Set to store IDs of listings that have already been scraped to avoid duplicates. This is a
+    # temporary solution until a database is set up, which will then take care of duplicates etc
+    ALREADY_SCRAPED_LISTINGS = set()
+
+    # Instance of AirBnbURLBuilder to generate URLs for different search parameters
+    URL_BUILDER = AirBnbURLBuilder()
+
+    def start_requests(self):
+        """
+       Generates initial requests for scraping Airbnb listings using coordinates from a JSON file.
+
+       This function performs the following steps:
+       1. Reads already scraped Airbnb listing IDs from a CSV file and stores them in a set.
+       2. Reads coordinates from a JSON file and shuffles them.
+       3. Constructs and returns a list of initial Scrapy requests using template URLs and the read coordinates.
+
+       :return: List of Scrapy FormRequest objects initialized with map coordinates for Vancouver.
+        """
+        file_name = self.settings.get("CSV_STORE_FILE_NAME")
+        try:
+            with open(
+                    f"{file_name}",
+                    "r",
+                    encoding="utf8") as file:
+                read_csv_file = csv.DictReader(file)
+                for row in read_csv_file:
+                    row: dict
+                    ListingsSpider.ALREADY_SCRAPED_LISTINGS.add(row['airbnb_listing_id'])
+        except FileNotFoundError as e:
+            print(e)
+        requests = []
+        try:
+            with open('cordinates.json', 'r') as file:
+                coordinates = json.load(file)
+                urls = ListingsSpider.URL_BUILDER.get_urls("july", "june", "august")
+                if coordinates is not None:
+                    requests = [scrapy.FormRequest(
+                        url.format(cord["ne_lat"], cord["ne_lng"], cord["sw_lat"], cord["sw_lng"],
+                                   ZOOM_LEVEL,
+                                   ZOOM_LEVEL)) for url in urls for
+                        cord in
+                        coordinates]
+        except FileNotFoundError:
+            print("The file was not found")
+        except json.JSONDecodeError:
+            print("Error decoding JSON")
+        return requests
+
+    async def parse(self, response: Response, **kwargs):
+        """
+        Parse method for extracting listings from the Airbnb search results page.
+
+        Args:
+            response (Response): The response object from the request.
+
+        Yields:
+            Request: A request object for each listing's details page.
+        """
+        if response.status != 200:
+            return
```

Comment:
raise exception instead?
does not agree with docstring description

Reply:
I think this code is unreachable. Will likely remove it.